### DISCUSSION TITLE
Ignore URL for service discovery when deployed in kubernetes

### DISCUSSION THREAD

**Discussion by luuk777w** [microsoft/aspire#3703] [Q&A] [answered]
I have the following section in an `appsettings.json` of one of my services:

```json
  "NextCloudSettings": {
    "WebDavUrl": "https://url.to.server.gr/remote.php/dav/",
    "UserName": "username",
    "Password": "password"
  }
```

But now Aspire tries to resolve the service behind the URL, which won't work because it is an external (meaning, outside of the system) service. In preview-4 this did not happen, but since preview-5 it tries to resolve the service behind this one, resulting in the following error:

```json
{
    "status": "Internal Server Error!",
    "code": 500,
    "reason": "Name or service not known (url.to.server.gr:443)",
    "note": "See application log for stack trace."
}
```

> fail: Microsoft.AspNetCore.Diagnostics.ExceptionHandlerMiddleware[1]
      An unhandled exception has occurred while executing the request.
      System.Net.Http.HttpRequestException: Name or service not known (url.to.server.gr:443)

Is there any way I can make Aspire ignore this URL?

----

Edit: apparently, this only happens when the application is deployed in my kubernetes cluster, and not in development. I am using Aspir8 to generate

**Comment by davidfowl** [comment]
It's unclear from the details in this issue what the error is and what the moving pieces are to be able to help you diagnose the issue. You must have code in your application that is reading the configuration section and using it to set values on an http client.

What does the code consuming this configuration section look like?

**Reply by luuk777w** [reply]
I agree my question is pretty bare bones and does not provide a lot of detail. But this is partially because I do not know or understand where the issue lies. I will provide my code here, but I don't really think this helps a lot, considering you already assumed this. 

Besides: it is working fine in development, just not when it is deployed in Kubernetes. Therefore, I doubt that it has so much to do with my code, and is more a configuration issue. I use Aspir8 to generate the Kubernetes files, so maybe the issue lies there.

Below is the class and method where the NextCloud api is called. I am using a nuget package called WebDav.Client, which uses an HttpClient to make request to NextCloud. This uses the WebDavUrl from my appsettings. 

```csharp
public class NextCloudService : IDataStorageService
{
    private readonly IConfiguration _configuration;
    private readonly IWebDavClient _webDavClient;
    private readonly HttpClient _httpClient;

    private readonly string _userName;
    private readonly string _password;

    public NextCloudService(IConfiguration configuration)
    {
        _configuration = configuration;
        _httpClient = new HttpClient();

        var settings = _configuration.GetSection("NextCloudSettings");
        var url = settings["WebDavUrl"];
        _userName = settings["UserName"]!;
        _password = settings["Password"]!;

        _httpClient.BaseAddress = new Uri(url!);

        // WebDav.Client pacakge, but I don't believe the issue lies here
        _webDavClient = new WebDavClient(_httpClient);
    }

    public async Task<ResourcesResult> FindResourcesAsync(string directory)
    {
       // stripped down version of this method for briefety.
        var resourceResult = new ResourcesResult();
        var result = await _webDavClient.Propfind(directory);
        // do something with result here and return
        return resourceResult;
    }

```

**Comment by luuk777w** [comment, marked as answer]
I turned out to be DNS related on my VM. Sorry for asking here. Just happened to be an issue that occurred during the upgrade to preview 5 hence I thought it was Aspire related
